amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestRecordProcessor.java

```diff
@@ -60,6 +60,10 @@ public void processRecords(ProcessRecordsInput processRecordsInput) {
                 recordValidator.add(recordValidatorKey, data);
             }
 
+            processRecordsInput.checkpointer().checkpoint();
```

Comment:
Curious why this wasn't there already. Do only the new tests require this?

Reply:
Yes, only for the new tests. Existing tests do not verify the content in lease table before shutting down the worker.